src/parser/mod.rs

```diff
@@ -8287,13 +8287,36 @@ impl<'a> Parser<'a> {
         };
 
         let group_by = if self.parse_keywords(&[Keyword::GROUP, Keyword::BY]) {
-            if self.parse_keyword(Keyword::ALL) {
-                GroupByExpr::All
+            let expressions = if self.parse_keyword(Keyword::ALL) {
+                None
             } else {
-                GroupByExpr::Expressions(self.parse_comma_separated(Parser::parse_group_by_expr)?)
+                Some(self.parse_comma_separated(Parser::parse_group_by_expr)?)
+            };
+
+            let mut modifiers = vec![];
+            loop {
+                if dialect_of!(self is ClickHouseDialect | GenericDialect)
```

Comment:
The dialect check can live outside the loop? since it never changes, and we would let us avoid entering the loop for non-clickhouse dialects